### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by marcelomaidden**
## Filter
At this pull request, I add the filter capability to the application:
- Adds CategoryFilter component
- Adds changeFilter and changeVisibility actions
- Adds filterReducer responsible to change the string with either the category or All
- Changes booksReducer adding the ability to choose which Books will be shown based on the filter

### CODE REVIEW

File: src/components/CategoryFilter.js
Comment (teekaytech) [inline src/components/CategoryFilter.js]: This function (handleFilter) should be implemented in the BookList container component and passes down to the CategoryFilter component ([screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png))

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello @marcelomaidden :wave:

__You've made laudable progress in this milestone.__
- All linter checks are successful :100: 
- Great PR title and summary :+1: 

__However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] In the `Setup` section of your Readme, kindly add instruction(s) on how to install your project dependencies (i.e. npm install).
- [x] You're having some error in the console. Please fix all logged errors.
console


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

File: src/components/Book.js
Comment (uryelah) [inline src/components/Book.js:5]: - Nice use of ternary operators!

File: src/containers/BookList.js
Comment (uryelah) [inline src/containers/BookList.js:55]: - [ ] [OPTIONAL] If you want to learn the latest, and simplest, way of handling the connection of components with redux actions and states, check the documentation on [Redux hooks](https://react-redux.js.org/api/hooks) and [Redux ToolKit](https://redux-toolkit.js.org/).

**Review by uryelah — APPROVED** [review]
Hi **Marcelo Araújo**, I'm Sarah your reviewer for this PR 👩‍🚀 

You did a great job fixing the errors found by the last TSE and implementing a very well organized and divided redux code in this milestone. 

Go ahead and good luck with the next milestone! 🚀  

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**

---
    
    
**Keep being an awesome dev and good luck with your next milestone!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: src/containers/BookList.js
Comment (marcelomaidden) [reply src/containers/BookList.js:55]: Thanks for the review and all suggestions uryelah.

Comment (marcelomaidden) [reply src/containers/BookList.js:55]: Thanks for the review and all suggestions uryelah.

**Pull request merged**
